src/context/HicetnuncContext.js

```diff
@@ -221,7 +210,9 @@ class HicetnuncContextProviderClass extends Component {
 
         let objkts = await Tezos.wallet.at(this.state.objkts)
         await Tezos.wallet.at(this.state.v2).then(c => console.log(c.parameterSchema.ExtractSignatures()))
-        let marketplace = await Tezos.wallet.at(this.state.v2)
+        let marketplace = await Tezos.wallet.at(swap.contract_address) // this can be either v1, v2 or teia
+        let marketplaceTeia = await Tezos.wallet.at(MARKETPLACE_CONTRACT_TEIA) // we want the reswap happen on teia
```

Comment:
We should move this outside in the future to avoid calling it every time we need to call those contracts

Reply:
true, could be lazy-loaded and then cached.